SwiftHelpers/SHDatePickerController.swift

```diff
@@ -0,0 +1,199 @@
+        let constraints = hConstraints + vConstraints
+        view.addConstraints(constraints)
+
+        bottomView.backgroundColor = .whiteColor()
+    }
+
+    private func setupDatePicker() {
+        bottomView.addSubview(datePicker)
+        datePicker.translatesAutoresizingMaskIntoConstraints = false
+        let hConstraints = NSLayoutConstraint.constraintsWithVisualFormat("H:|[datePicker]|", options: NSLayoutFormatOptions(), metrics: nil, views: ["datePicker": datePicker])
+        let vConstraints = NSLayoutConstraint.constraintsWithVisualFormat("V:[datePicker]|", options: NSLayoutFormatOptions(), metrics: nil, views: ["datePicker": datePicker])
+        let constraints = hConstraints + vConstraints
+        view.addConstraints(constraints)
+    }
+
+    private func setupToolbar() {
+        bottomView.addSubview(toolbar)
+        toolbar.translatesAutoresizingMaskIntoConstraints = false
+        let hConstraints = NSLayoutConstraint.constraintsWithVisualFormat("H:|[toolbar]|", options: NSLayoutFormatOptions(), metrics: nil, views: ["toolbar": toolbar])
+        let vConstraints = NSLayoutConstraint.constraintsWithVisualFormat("V:|[toolbar][datePicker]", options: NSLayoutFormatOptions(), metrics: nil, views: ["toolbar": toolbar, "datePicker": datePicker])
+        let constraints = hConstraints + vConstraints
+        view.addConstraints(constraints)
+
+        let leftSpace = UIBarButtonItem(barButtonSystemItem: .FixedSpace, target: nil, action: nil)
+        leftSpace.width = 10
+        let cancelButton = UIBarButtonItem(barButtonSystemItem: .Cancel, target: self, action: #selector(tappedCancelButton(_:)))
+        let centerSpace = UIBarButtonItem(barButtonSystemItem: .FlexibleSpace, target: nil, action: nil)
+        let confirmButton = UIBarButtonItem(barButtonSystemItem: .Done, target: self, action: #selector(tappedDoneButton(_:)))
+        let rightSpace = UIBarButtonItem(barButtonSystemItem: .FixedSpace, target: nil, action: nil)
+        rightSpace.width = 10
+        toolbar.items = [leftSpace, cancelButton, centerSpace, confirmButton, rightSpace]
+    }
+
+    func tappedDoneButton(button: UIBarButtonItem) {
+        dismissWithDate(datePicker.date)
+    }
+
+    func tappedCancelButton(button: UIBarButtonItem) {
+        dismissWithDate(nil)
+    }
+
+    func tappedBackgroundView(gestureRecognizer: UITapGestureRecognizer) {
+        dismissWithDate(nil)
+    }
+
+    func swipedBackgroundView(gestureRecognizer: UISwipeGestureRecognizer) {
+        dismissWithDate(nil)
+    }
+
+    private func dismissWithDate(date: NSDate?) {
+        dismissViewControllerAnimated(true) {
+            self.completion?(date)
+        }
+    }
+
+}
+
+/// A class used to animate the transition of a SHDatePickerViewController
+public class SHDatePickerAnimatedTransition: NSObject, UIViewControllerAnimatedTransitioning {
+
+    private let isPresenting: Bool
```

Comment:
I know in lots of examples they use two instances for presenting and dismissing, but I used to just do instance.presenting = true/false before returning the animator object. Not a big deal :)